src/Core.hs

```diff
@@ -63,14 +63,22 @@ data TypePattern
   | AnyType Ident Var
   deriving (Data, Eq, Show)
 
-data ConstructorPattern
-  = ConstructorPattern !Constructor [(Ident, Var)]
-  | AnyConstructor Ident Var
+data ConstructorPatternF pat
+  = CtorPattern !Constructor [pat]
+  | PatternVar Ident Var
+  deriving (Data, Eq, Foldable, Functor, Show, Traversable)
+makePrisms ''ConstructorPatternF
+
+newtype ConstructorPattern =
+  ConstructorPattern { _constructorPattern :: ConstructorPatternF ConstructorPattern }
   deriving (Data, Eq, Show)
-makePrisms ''ConstructorPattern
+makeLenses ''ConstructorPattern
+
+instance Phased a => Phased (ConstructorPatternF a) where
+  shift i = fmap (shift i)
 
 instance Phased ConstructorPattern where
-  shift _ = id
+  shift i = over constructorPattern (shift i)
```

Comment:
That's just a more complicated version of `id`, no?

Reply:
It's a more future-proof one. If the thing inside becomes shiftable, then this will do it.